### ISSUE TITLE
Lint/FormatParameterMismatch false positive for fields with absolute argument numbers

### ISSUE THREAD

**Issue by 0leynikov** [rubocop-hq/rubocop#6482]
In the format string for `Kernel#format` you can specify the absolute argument number before a field using `(digit)$` flag. Here are examples from the [docs](https://ruby-doc.org/core-2.5.3/Kernel.html#method-i-format):

```ruby
sprintf("%1$*2$s %2$d %1$s", "hello", 8)   #=> "   hello 8 hello"
sprintf("%1$*2$s %2$d", "hello", -8)       #=> "hello    -8"
```

RuboCop currently fails to deal with such situations.

--------

## Expected behavior

No offenses.

## Actual behavior

Lint/FormatParameterMismatch offense.

## Steps to reproduce the problem

Given this minimal example:

```ruby
# format.rb
format('%1$s %1$s', 'foo')
```
Run:
```
$ rubocop format.rb
Inspecting 1 file
W

Offenses:

format.rb:1:1: W: Lint/FormatParameterMismatch: Number of arguments (1) to format doesn't match the number of fields (0).
format('%1$s %1$s', 'foo')
^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.5.1 x86_64-openbsd)
```

**Comment by koic** [comment]
Thanks for the feedback. I confirmed this reproduction and I opened a PR #6483.

**Comment by 0leynikov** [comment]
Whoa, that's fast! Thank you.
